make docker-build requires tags

When running `make docker-build` from a forked repo it will fail. 
```
make docker-build
fatal: No names found, cannot describe anything.
GOOS=linux GOARCH=amd64 GOPATH= DATE=2023-05-31 COMMIT=aea43655 VERSION= \
	  /Users/amcohen@us.ibm.com/Projects/cloudnative-pg/bin/goreleaser build --skip-validate --clean --single-target
  • starting build...
  • loading config file                              file=.goreleaser.yml
  • building only for linux/amd64                    reason=single target is enabled
  • loading environment variables
    • using token from "$GITHUB_TOKEN"
  • getting and validating git state
  ⨯ build failed after 0s                    error=git doesn't contain any tags. Either add a tag or use --snapshot
make: *** [docker-build] Error 1
```

The error is thrown by goreleaser wanting tags. 
```
 goreleaser release --debug
  • debug logs enabled
  • starting release...
  • loading config file                              file=.goreleaser.yml
  • loaded config file                               config={cnpg [GO111MODULE=on CGO_ENABLED=0] {cloudnative-pg/cloudnative-pg   false false      [] []    } [] [] [] [{[] cnpg {    {  } {false }} { }   Manage your CloudNativePG clusters This plugin provides multiple commands to help you manage your CloudNativePG clusters. https://github.com/cloudnative-pg/cloudnative-pg    true}] [] { [] {    {  } {false }}  { }      []  [] [] [] [] } [] [{manager [linux] [amd64 arm64] [] [] [] [] []  cmd/manager/main.go manager/manager_{{ .Arch }} {[] []}   false   true false   { [-s -w -X github.com/cloudnative-pg/cloudnative-pg/pkg/versions.buildVersion={{.Env.VERSION}} -X github.com/cloudnative-pg/cloudnative-pg/pkg/versions.buildCommit={{.Env.COMMIT}} -X github.com/cloudnative-pg/cloudnative-pg/pkg/versions.buildDate={{.Env.DATE}}] [] [] [] [all=-trimpath={{.Env.GOPATH}};{{.Env.PWD}}] []} []} {kubectl-cnpg [darwin linux windows] [amd64 386 arm64 arm ppc64le s390x] [5 6 7] [] [] [] [{darwin 386   } {windows ppc64le   } {windows s390x   }]  cmd/kubectl-cnpg/main.go kubectl-cnpg {[] []}   false   false false   { [-s -w -X github.com/cloudnative-pg/cloudnative-pg/pkg/versions.buildVersion={{.Env.VERSION}} -X github.com/cloudnative-pg/cloudnative-pg/pkg/versions.buildCommit={{.Env.COMMIT}} -X github.com/cloudnative-pg/cloudnative-pg/pkg/versions.buildDate={{.Env.DATE}}] [] [] [] [all=-trimpath={{.Env.GOPATH}};{{.Env.PWD}}] []} []}] [{ [kubectl-cnpg] {  ----------  0001-01-01 00:00:00 +0000 UTC} kubectl-cnpg_{{ .Version }}_ {{- .Os }}_ {{- if eq .Arch "amd64" }}x86_64 {{- else if eq .Arch "386" }}i386 {{- else }}{{ .Arch }}{{ with .Arm }}v{{ . }}{{ end }}{{ end }} map[]  []  false false [] false false}] [{{kubectl-cnpg_{{ .Version }}_ {{- .Os }}_ {{- if eq .Arch "amd64" }}x86_64 {{- else if eq .Arch "386" }}i386 {{- else }}{{ .Arch }}{{ with .Arm }}v{{ . }}{{ end }}{{ end }}      map[] [] [] [] [] [] [] [] {   } {   { } { }} {{ } {[] [] [] [] [] []} [] {  } []} {{ } {  }} {  { }}} map[] kubectl-cnpg [kubectl-cnpg] [rpm deb]    https://github.com/cloudnative-pg/cloudnative-pg Marco Nenciarini <marco.nenciarini@enterprisedb.com>   /usr/local/bin  false}] [] {{{ .Tag }}-next} {{{ .ProjectName }}-{{ .Version }}-checksums.txt  [] false []} [] [] [] [] [] [] {{[]}  true  [] 0}  [{  [--batch -u {{ .Env.GPG_FINGERPRINT }} --output ${signature} --detach-sign ${artifact}]  checksum [] <nil>  []  false}] [] {  } {[go mod download]} {  false  [] false} {false []  } { {false } {false  } {false     } {false      [] []} {false    } {false    } {false  0   []   false} {false       } {false } {false  } {false false   map[] } {false   }} [] [] { } false [] [] { [] [] [] [] [] [] []    {[] []}   false   false false   { [] [] [] [] [] []} []}  {   false} {  false false false} {  false}}
  • parallelism: 16
  • loading environment variables
    • using token from "$GITHUB_TOKEN"
    • token type: github
  • getting and validating git state
    • running git                                    args=[-c log.showSignature=false rev-parse --is-inside-work-tree]
    • git result                                     stdout=true
 stderr=
    • running git                                    args=[-c log.showSignature=false rev-parse --is-inside-work-tree]
    • git result                                     stdout=true
 stderr=
    • running git                                    args=[-c log.showSignature=false rev-parse --abbrev-ref HEAD --quiet]
    • git result                                     stdout=main
 stderr=
    • running git                                    args=[-c log.showSignature=false show --format=%h HEAD --quiet]
    • git result                                     stdout=aea43655
 stderr=
    • running git                                    args=[-c log.showSignature=false show --format=%H HEAD --quiet]
    • git result                                     stdout=aea436555e29102f2509e0140e7ed8642ac134c3
 stderr=
    • running git                                    args=[-c log.showSignature=false rev-list --max-parents=0 HEAD]
    • git result                                     stdout=f036736296cd44fc5911479ca9f4b3ac76ee12ca
 stderr=
    • running git                                    args=[-c log.showSignature=false show --format='%ct' HEAD --quiet]
    • git result                                     stdout='1685549369'
 stderr=
    • running git                                    args=[-c log.showSignature=false describe --always --dirty --tags]
    • git result                                     stdout=aea43655
 stderr=
    • running git                                    args=[-c log.showSignature=false ls-remote --get-url]
    • git result                                     stdout=git@github.com:aaronmcohen/cloudnative-pg.git
 stderr=
    • running git                                    args=[-c log.showSignature=false tag --points-at HEAD --sort -version:refname]
    • git result                                     stdout= stderr=
    • running git                                    args=[-c log.showSignature=false describe --tags --abbrev=0 HEAD]
    • git result                                     stdout= stderr=fatal: No names found, cannot describe anything.

  ⨯ release failed after 0s                  error=git doesn't contain any tags. Either add a tag or use --snapshot
```

# Workaround

By default a fork does not include the tags from the forked repo. To add them run:
- git remote add upstream git@github.com:cloudnative-pg/cloudnative-pg.git
- git fetch --tags upstream
- git push -f --tags origin main

--- a/Makefile
+++ b/Makefile
@@ -120,8 +120,14 @@ run: generate fmt vet manifests ## Run against the configured Kubernetes cluster
 	go run ./cmd/manager
 
 docker-build: go-releaser ## Build the docker image.
-	GOOS=linux GOARCH=${ARCH} GOPATH=$(go env GOPATH) DATE=${DATE} COMMIT=${COMMIT} VERSION=${VERSION} \
-	  $(GO_RELEASER) build --skip-validate --clean --single-target
+	if [ -z "${VERSION}" ] ; \
+	then \
+	  GOOS=linux GOARCH=${ARCH} GOPATH=$(go env GOPATH) DATE=${DATE} COMMIT=${COMMIT} VERSION=${VERSION} \
+		$(GO_RELEASER) build --clean --single-target --snapshot ;\
+	else \
+	  GOOS=linux GOARCH=${ARCH} GOPATH=$(go env GOPATH) DATE=${DATE} COMMIT=${COMMIT} VERSION=${VERSION} \
+		$(GO_RELEASER) build --skip-validate --clean --single-target ;\
+	fi ;\
 	DOCKER_BUILDKIT=1 docker build . -t ${CONTROLLER_IMG} --build-arg VERSION=${VERSION}
 
 docker-push: ## Push the docker image.

--- a/Makefile
+++ b/Makefile
@@ -120,14 +120,8 @@ run: generate fmt vet manifests ## Run against the configured Kubernetes cluster
 	go run ./cmd/manager
 
 docker-build: go-releaser ## Build the docker image.
-	if [ -z "${VERSION}" ] ; \
-	then \
-	  GOOS=linux GOARCH=${ARCH} GOPATH=$(go env GOPATH) DATE=${DATE} COMMIT=${COMMIT} VERSION=${VERSION} \
-		$(GO_RELEASER) build --clean --single-target --snapshot ;\
-	else \
-	  GOOS=linux GOARCH=${ARCH} GOPATH=$(go env GOPATH) DATE=${DATE} COMMIT=${COMMIT} VERSION=${VERSION} \
-		$(GO_RELEASER) build --skip-validate --clean --single-target ;\
-	fi ;\
+	GOOS=linux GOARCH=${ARCH} GOPATH=$(go env GOPATH) DATE=${DATE} COMMIT=${COMMIT} VERSION=${VERSION} \
+	  $(GO_RELEASER) build --skip-validate --clean --single-target $(if $(VERSION),,--snapshot)
 	DOCKER_BUILDKIT=1 docker build . -t ${CONTROLLER_IMG} --build-arg VERSION=${VERSION}
 
 docker-push: ## Push the docker image.

